src/cmap/connection.ts

```diff
@@ -746,12 +747,17 @@ export class Connection extends TypedEventEmitter<ConnectionEvents> {
         }
       }
     } catch (readError) {
+      const err = readError;
       if (TimeoutError.is(readError)) {
-        throw new MongoOperationTimeoutError(
+        const error = new MongoOperationTimeoutError(
           `Timed out during socket read (${readError.duration}ms)`
         );
+        this.dataEvents = null;
+        this.onError(error);
+        throw error;
       }
-      throw readError;
+      this.cleanup(err);
```

Comment:
This isn't correct right? it should just be the clean up inside the timeout if block